src/pspm_find_valid_fixations.m

```diff
@@ -153,312 +148,248 @@
+                  end
+             end
+             if ~strcmpi(y_unit,'mm')
+                [nsts,y_data] = pspm_convert_unit(gaze_y.data, y_unit, 'mm');
+                [msts,y_range] = pspm_convert_unit(transpose(gaze_y.header.range), y_unit, 'mm');
+                  if nsts~=1 || msts~=1
+                    warning('ID:invalid_input', 'Failed to convert data.');
+                    return
+                  end
+             end            
+         
+            % convert normalized fixation points to data resolution
+            fix_point_temp = zeros(size(fix_point));
             fix_point_temp(:,1) = x_range(1)+ fix_point(:,1)* diff(x_range);
             fix_point_temp(:,2) = y_range(1)+ fix_point(:,2)* diff(y_range);
 
-            % calculate the middlepoint of the display
-            middlepoint= [x_range(1)+ diff(x_range)/2, ...
-              y_range(1)+ diff(y_range)/2];
-
-            % caluculate the visual angle of the fixation points
-            % according to the right range
-
-            dist = middlepoint - fix_point_temp;
-            dist = sqrt(dist(:,1).^2 + dist(:,2).^2);
-            angle_of_fix = 2 * atan(dist/(2*distance));
-            angle_of_fix = rad2deg(angle_of_fix);
-
-            % find for each fixation point the right radius
-            tot_angle = angle_of_fix + circle_degree;
-            tot_angle = deg2rad(tot_angle);
-            radius = 2*distance * tan(tot_angle/2);
-            radius = radius - dist;
-
-            % calculate for ech point distance to fixationpoint
-            gaze_data = [x_data,y_data];
-            dist_fix_gaze = fix_point_temp - gaze_data;
-            dist_fix_gaze = (sqrt(dist_fix_gaze(:,1).^2 + dist_fix_gaze(:,2).^2));
-
+            % calculate the visual angle of the gaze points
+            gaze_dist = fix_point_temp - [x_data(:), y_data(:)];
+            gaze_dist = sqrt(gaze_dist(:,1).^2 + gaze_dist(:,2).^2);
+            angle_of_gaze = 2 * atan(gaze_dist/(2*distance));
+           
             % compare calculated distance to accepted radius
-            excl = dist_fix_gaze > radius;
+            excl = angle_of_gaze > deg2rad(circle_degree);
 
+            % check plotting
             if options.plot_gaze_coords
               fg = figure;
               ax = axes('NextPlot', 'add');
               set(ax, 'Parent', handle(fg));
 
-              % validation middlepoint
+              % first fixation point
               x_point = fix_point_temp(1,1);
               y_point = fix_point_temp(1,2);
+              radius  = tan(deg2rad(circle_degree)/2) * 2 * distance;
 
-              %for the circle around the first fixation point
+              % plot the circle around the first fixation point
               th = 0:pi/50:2*pi;
               x_unit = radius(1) * cos(th) + x_point;
               y_unit = radius(1) * sin(th) + y_point;
 
               % plot gaze coordinates
               mi=min(min(x_data),min(y_data));
```

Comment:
If the `axis` operation is removed, the `mi` and `ma` calculations seem to be removable too?

Reply:
`axis`was commented for testing purposes, it is uncommented now.